docs/user-guide.md

```diff
@@ -0,0 +1,51 @@
+# User Guide
+
+## Building the base and node images
+### Base image
+`kind` runs a local kubernetes cluster by using Docker containers as "nodes."
+`kind` uses the [`node-image`][node image] to run kubernetes artifacts, such 
+as `kubeadm` or `kubelet`.
+The `node-image` in turn is built off the [`base-image`][base image], which
+installs all the dependencies needed for Docker and Kubernetes to run in a
+container.
+
+To build the `base-image` we use the `build` command:
```

Comment:
Users generally shouldn't need to build the base image, as we've published a pre-built one, we should probably detail this in a more advanced guide and focus on something like:

---
# User Guide
// what kind is
## Creating a Cluster
// how to create a generic cluster
// note about using a custom image
## Building Images
// how to build the node image, note about the base image existing

## Advanced
### Building The Base Image
// how to build the base image, how to use it with the node image
### Using Config
// how to use a config file and e.g. patch the kubeadm config

----

That way we can focus on the easy and most relevant parts first and progressively get into the weeds on implementation details and advanced usage. Most users don't need to worry too much about the base image, just know that it exists.

Reply:
actually I'll rearrange a bit in a follow up